com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1463,6 +1461,31 @@ private void HandleRawTransportPoll(NetworkEvent networkEvent, ulong clientId, A
             }
         }
 
+        /// <summary>
+        /// Handles cleaning up the transport id/client id tables after
+        /// receiving a disconnect event from transport
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private ulong TransportIdCleanUp(ulong clientId, ulong transportId)
+        {
+            // This check is for clients that attempted to connect but failed.
+            // When this happens, the client will not have an entry within the
+            // m_TransportIdToClientIdMap or m_ClientIdToTransportIdMap lookup
+            // tables so we exit early and just use the transport clientId
+            // for the disconnect event.
+            if (!IsServer && !m_TransportIdToClientIdMap.ContainsKey(clientId))
+            {
+                return clientId;
```

Comment:
`return 0;` ? That would be justas valid as we don't really have a clientId here.